ipatests/test_integration/test_sssd.py

```diff
@@ -180,3 +184,60 @@ def test_extdom_group(self):
             client_conf_backup.restore()
             tasks.clear_sssd_cache(self.master)
             tasks.clear_sssd_cache(client)
+
+    def test_external_group_paging(self):
+        """SSSD should fetch external groups without any limit.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/4058
+        1: Add external groups more than limit.
+        2: Run the command id aduser@ADDOMAIN.COM
+        3: sssd should retrieve all the external groups.
+        """
+        cmd = self.master.run_command(['sssd', '--version'])
+        sssd_version = platform_tasks.parse_ipa_version(
+            cmd.stdout_text.strip())
+        if LooseVersion(sssd_version.version) <= LooseVersion('2.2.0'):
```

Comment:
1. Do you really need LooseVersion?
2. the fix is available in sssd 2.2.2

Reply:
1. sssd_version <= platform_tasks.parse_ipa_version('1.16.3'): this will compare string with a string.
That's why it needs LooseVersion, as sssd_version.version with type string.
2. In PR-CI with f-29 is using version 2.2.0